pallets/protos/src/lib.rs

```diff
@@ -455,6 +455,37 @@ pub mod pallet {
 			Ok(())
 		}
 
+
+		/// Delete Proto-Fragment `proto_hash` from all relevant Storage Items
+		#[pallet::weight(50_000)]
+		pub fn delete_proto(origin: OriginFor<T>, proto_hash: Hash256) -> DispatchResult {
+			ensure_root(origin)?;
+
+			let proto_struct = <Protos<T>>::get(&proto_hash).ok_or(Error::<T>::ProtoNotFound)?;
+
+			<ProtosByCategory<T>>::mutate(&proto_struct.category, |list_protos| {
+				if let Some(list_protos) = list_protos {
+					list_protos.retain(|current_hash| proto_hash != *current_hash);
+				}
+			});
+			<ProtosByOwner<T>>::mutate(proto_struct.owner, |list_protos| {
+				if let Some(list_protos) = list_protos {
+					list_protos.retain(|current_hash| proto_hash != *current_hash);
+				}
+			});
+			for account_that_curated_proto  in <ProtoCurations<T>>::iter_key_prefix(proto_hash) {
+				<AccountCurations<T>>::mutate(&account_that_curated_proto, |list_protos| {
+					if let Some(list_protos) = list_protos {
+						list_protos.retain(|current_hash| proto_hash != *current_hash);
+					}
+				});
+			}
+			<ProtoCurations<T>>::clear_prefix(proto_hash, u32::MAX, None);
```

Comment:
Full disclosure: I don't know what the last parameter `maybe_cursor` of the function `clear_prefix()` does 😅  @sinkingsugar .

So this might a failure point for this extrinsic

Reply:
hopefully should work since all relevant storage items have been cleared

Reply:
Ah it's simply cos if there is too much to delete it might not fit one block so you spread deletion in many blocks...

Reply:
Should be ok for now like you did

Reply:
`remove_prefix` is deprecated though? We should still use it?

Reply:
Ah, no then 👍